db/migrate/20240517220008_add_moved_after_payments_frozen_at_cohort_to_participant_profile.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class AddMovedAfterPaymentsFrozenAtCohortToParticipantProfile < ActiveRecord::Migration[7.1]
+  disable_ddl_transaction!
+
+  def change
+    add_column :participant_profiles, :moved_after_payments_frozen_at_cohort, :integer, limit: 2
```

Comment:
I think the naming here is hard to understand. 
This might be because we're attempting to describe the reason for moving from this cohort in the column name.
It might be more flexible to have `previous_cohort` as a foreign key or integer and `cohort_change_reason` as a text or enum description.

Reply:
The thing is that we only keep track of these specific cohort changes.
I mean, we are not interested in tracking the previous_cohort for other cohort changes we do but for this specific type one.

Reply:
Yes, there are other reasons why we move cohort but we are not caring about the reason.
In any case, I'm happy to add a reason field if that clarifies things for devs.

Reply:
Throwing an idea out there... would adding a table for this be a better idea?  The table could join participant_profile, cohort and a reason?  It could be more generic and we could include _any_ time we move a participant to a different cohort?  It would be useful for being able to track how many people we've moved because of induction dates/temp cohort placement because we do get asked about numbers for that kind of thing too?

Reply:
I agree with @steventux that the naming is difficult to understand, appreciate it's difficult to name though, so that for me is another reason a separate table with the reason might be easier to track these. Much easier to give it a nice reason rather than awkward column naming?

Reply:
We can do it but the issue is that if we do, again we are persisting several times the same fact (therefore, data inconsistencies is a matter of time).

Given that we already have induction records that track any change in the induction facts of a participant, I think the best place to flag this is there. That is, every time we change the cohort of a participant we are adding a new induction record to the chain pointing to the new record. So to me, just adding a flag field there stating it is because of previous cohort froze payments is more than enough and no duplication of data persisted.

However, the LPDOB team think that querying the induction records is a pain and adding more fields make it awful for them. So they suggested to put the field into the ParticipantProfile. 
However by pointing the ParticipantProfile to the previous cohort we have persisted twice the cohort change fact: 
   - The previous IR pointing to the previous cohort and the new IR pointing to the new cohort means cohort change.
   - The field in ParticipantProfile pointing to the previous cohort also means cohort change.

These kind of data duplication create inconsistencies sooner or later and I dont' like them at all.

Adding a table would put the changes in its own model but again we would be persisting twice the same facts and therefore very likely to add inconsistencies again.

I'm open to any solution to agree though.

Reply:
Would a foreign key (instead of an arbitrary integer column) also help to solve the problem of data duplication and the risk of corruption as you mention @ltello ? This way we always point to a valid Cohort record.

Reply:
My thought was the separate table would effectively be a history of changes made to a participants cohort and the reason why. Also assuming there was a `created_at` timestamp we'd see the order of them too.  I'm happy either way, I think the addition of the reason is a good call but of course we can only persist the latest cohort move on the participant profile.

Reply:
@ethax-ross what's your view on this?